src/Lean/Elab/Deriving/FromToJson.lean

```diff
@@ -28,7 +28,7 @@ def mkToJsonInstanceHandler (declNames : Array Name) : CommandElabM Bool := do
         let fields := getStructureFieldsFlattened (← getEnv) declNames[0] (includeSubobjectFields := false)
         let fields : Array Syntax ← fields.mapM fun field => do
           let (isOptField, nm) := mkJsonField field
-          if isOptField then `(opt $nm $(mkIdent <| header.targetNames[0] ++ field))
+          if isOptField then `($(mkIdent ``Json.opt) $nm (some $(mkIdent <| header.targetNames[0] ++ field)))
```

Comment:
This looks wrong; if `opt` is in scope, this shouldn't be necessary

Reply:
Correct lexical binding (i.e. "hygiene") requires that they are resolved at macro-definition site only

Reply:
Ah, I remembered now why I prefer the `mkIdent`. It's true that the name is resolved using macro-definition scopes, but it is resolved at macro-usage *time* since Lean does not run elaboration or any non-syntactic checks when defining the macro. So I am able to write down something incorrect there like `oppt`, and it will define fine but will fail at usage time. On the other hand ```mkIdent ``opt``` would fail thanks to the ``` `` ```. But I don't strongly prefer checking, I know it's correct in this case.